### ISSUE TITLE
pyenv-win cannot work on windows nushell

### ISSUE THREAD

**Issue by Salted-Fish-Swimming** [nushell/nushell#14583]
### Describe the bug

When I using `pyenv local`, this command will report a nushell-error `Could not access file '/c': ...` . It well report error when execute `pyenv` with any sub-command, except without sub-command.

### How to reproduce

1. Install pyenv-win on powershell ( it cannot work correctly on other shell)

```
Invoke-WebRequest -UseBasicParsing -Uri "https://raw.githubusercontent.com/pyenv-win/pyenv-win/master/pyenv-win/install-pyenv-win.ps1" -OutFile "./install-pyenv-win.ps1"; &"./install-pyenv-win.ps1"
```

2. log in nushell, run `pyenv local`

3. See error

### Expected behavior

When i use `pyenv` with some sub-command, it won't report any errors.

I'm not sure it is pyenv's bug or nushell caused bug, so I will also put a issue in pyenv-win .

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.100.0                                  |
| major              | 0                                        |
| minor              | 100                                      |
| patch              | 0                                        |
| branch             |                                          |
| commit_hash        | 30f98f7e64a63a159ce70f02aa6c4c1196cd9ca7 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.80.1 (3f5fd8dd4 2024-08-06)      |
| rust_channel       | 1.80.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.80.1 (376290515 2024-07-16)      |
| build_time         | 2024-11-13 03:41:33 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash                   |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
I can't tell what's failing, more detailed error messages would be helpful. It could be because pyenv-win looks like a mix of ancient technology of batch files and vb script files which call out to cmd and cscript.

**Comment by Salted-Fish-Swimming** [comment]
> I can't tell what's failing, more detailed error messages would be helpful. It could be because pyenv-win looks like a mix of ancient technology of batch files and vb script files which call out to cmd and cscript.

Sorry, i forget to post the error message image for report. It's here

![图片](https://github.com/user-attachments/assets/a30e3771-95bc-4065-a229-724d870a15fa)

Yes, `which pyenv` show it is a .bat file. [This is the file content](https://github.com/pyenv-win/pyenv-win/blob/master/pyenv-win/bin/pyenv.bat)

**Comment by fdncred** [comment]
Right, I already looked at it. That bat file calls the vbs file in libexec. The "could not access" errors are typically permission errors or something strange on the file system. It's going to take someone on windows to diagnose this and I don't plan on using windows this weekend.

**Comment by Salted-Fish-Swimming** [comment]
> Right, I already looked at it. That bat file calls the vbs file in libexec. The "could not access" errors are typically permission errors or something strange on the file system. It's going to take someone on windows to diagnose this and I don't plan on using windows this weekend.

It's not urgent

**Comment by Salted-Fish-Swimming** [comment]
if I use the following content,

```lua
-- ...
return {
  -- ...
  default_prog = {
    'C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\powershell.exe',
    '-Command', 'nu'
  },
 -- ...
}
```

instead of the following code in `wezterm.lua` as wezterm config file content.

```lua
-- ...
return {
  -- ...
  set_environment_variables = {
    COMSPEC =
      "C:\\Users\\31090\\AppData\\Local\\Programs\\nu\\bin\\nu.exe",
  },
  -- ...
}
```

It can be work.

It other words, if you use `powershell` , `cmd` or `git-bash` as the original shell to start up `nushell` , instead of directly starting up the `nushell`, it will be work.

**Comment by Salted-Fish-Swimming** [comment]
if I use the following content,

```lua
-- ...
return {
  -- ...
  default_prog = {
    'C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\powershell.exe',
    '-Command', 'nu'
  },
 -- ...
}
```

instead of the following code in `wezterm.lua` as wezterm config file content.

```lua
-- ...
return {
  -- ...
  set_environment_variables = {
    COMSPEC =
      "C:\\Users\\31090\\AppData\\Local\\Programs\\nu\\bin\\nu.exe",
  },
  -- ...
}
```

It can be work.

It other words, if you use `powershell` , `cmd` or `git-bash` as the original shell to start up `nushell` , instead of directly starting up the `nushell`, it will be work.
